src/compute/src/rpc/service/stream_service.rs

```diff
@@ -210,4 +131,27 @@ impl StreamService for StreamServiceImpl {
 
         Ok(Response::new(WaitEpochCommitResponse { status: None }))
     }
+
+    async fn streaming_control_stream(
+        &self,
+        request: Request<Streaming<StreamingControlStreamRequest>>,
+    ) -> Result<Response<Self::StreamingControlStreamStream>, Status> {
+        let mut stream = request.into_inner().boxed();
+        let first_request = stream
+            .try_next()
+            .await?
+            .ok_or_else(|| Status::invalid_argument("failed to receive first request"))?;
+        let StreamingControlStreamRequest {
+            request: Some(streaming_control_stream_request::Request::Init(init_request)),
+        } = first_request
+        else {
+            return Err(Status::invalid_argument(format!(
+                "unexpected first request: {:?}",
+                first_request
+            )));
+        };
```

Comment:
Seems like we can simplify this a little by matching:

```
stream
            .try_next()
            .await?
```

Then handle the following branches:
```
- Err(_) => ...
- Ok(StreamingControlRequest ...) => ...
- Ok(_) => ...
```